macros/recursive_dag.sql

```diff
@@ -1,14 +1,25 @@
-{% macro recursive_dag() %}
-    {{ return(adapter.dispatch('recursive_dag', 'dbt_project_evaluator')()) }}
+{% macro recursive_dag(filter_views=false) %}
+    {{ return(adapter.dispatch('recursive_dag', 'dbt_project_evaluator')(filter_views)) }}
 {% endmacro %}
 
-{% macro default__recursive_dag() %}
+{% macro default__recursive_dag(filter_views) %}
 
 with recursive direct_relationships as (
     select  
         *
     from {{ ref('int_direct_relationships') }}
+
+    {% if not filter_views %}
+    -- when not filtering on views, we want all nodes, except tests
     where resource_type <> 'test'
+    {% else %}
+    -- when filtering on views, we only want models and consider relationships that are
+    -- either views, ephemerals or sources
```

Comment:
The naming here is a bit confusing.. why for `filter_views` do we include `view`s, `ephemeral`s, and `source`s?

Reply:
maybe `filter_materialized` could be closer? i guess in dbt parlance materialized would also apply to views